onnxruntime/core/optimizer/nhwc_transformer.h

```diff
@@ -22,7 +22,11 @@ class NhwcTransformer : public GraphTransformer {
  public:
   explicit NhwcTransformer(AllocatorPtr cpu_allocator) noexcept 
     : GraphTransformer("NhwcTransformer"), cpu_allocator_(std::move(cpu_allocator)){};
-
+#ifdef USE_XNNPACK
+  //Can XNNPack run the conv node?
+  static bool IsConvSupportedByXNNPack(const Node& nodeRef, bool input_is_nchw);
+  static bool IsMaxPoolSupportedByXNNPack(const Node& nodeRef, bool input_is_nchw);
```

Comment:
Not really a fan of putting such XNNPack specific things here. It would be cleaner if we could split out any shared functionality and have a separate transformer for XNNPack.

Reply:
Currently it works as:
1. If an ONNX conv node can be run with xnnpack, convert it to NHWC format first.
2. Then, if an NHWC conv node can be run with xnnpack, convert it to xnnpack conv.

The idea is one day when we have NHWC conv in mlas, we can reuse this code and remove the IsConvSupportedByXNNPack test from this file. Surely I can combine these two steps to a single one, but it also means XNNPack graph transformer will be conflict with this NHWC transformer.